### PR TITLE
ddl: Fix adding vector index on empty table may blocked forever

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/9032

Problem Summary:

When creating vector index on empty table, the DDL may be blocked forever.

This is because when an empty table is synced to TiFlash, the `StorageDeltaMerge` is created without initializing the `DeltaMergeStore` instance. This is to avoid generating too many fragmented, small files that exhaust the filesystem inode.
However, if the `DeltaMergeStore` instance is not created, then `system.dt_local_index` will not get the local index stats. Which make tidb consider the vector index is still under building and the ddl does not end.

https://github.com/pingcap/tiflash/blob/1a781c7b0a34e7b87a47770cab5c169d3821ef54/dbms/src/Storages/System/StorageSystemDTLocalIndexes.cpp#L88-L94

### What is changed and how it works?

```commit-message

```

* If `StorageDeltaMerge` is created but `DeltaMergeStore` is not inited, then generate the local index info by the TiDB::TableInfo
  

Small refactor:
* Remove a verbose logging when system table is read by tidb-server

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/InterpreterSelectQuery.cpp [hunk 1]
```diff
@@ -243,7 +243,6 @@ void InterpreterSelectQuery::getAndLockStorageWithSchemaVersion(const String & d
             || (managed_storage->engineType() != ::TiDB::StorageEngine::DT
                 && managed_storage->engineType() != ::TiDB::StorageEngine::TMT))
         {
-            LOG_DEBUG(log, "{}.{} is not ManageableStorage", database_name, table_name);
```
Comment (JaySon-Huang) [inline dbms/src/Interpreters/InterpreterSelectQuery.cpp:246]: Running queries on the system table will trigger this logging, which is verbose but not provide any information. Remove it.

**Comment by Lloyd-Pottiger** [comment]
Why not just insert a row even when store is not init?

**Comment by breezewish** [comment]
Not sure whether this idea is ok, we may also regard progress=1 and missing table in tiflash_indexes as index build is finished, because progress=1 already means TiFlash is aware of this table, and then no entry in tiflash_indexes means exactly this case?

**Comment by JaySon-Huang** [comment]
> Not sure whether this idea is ok, we may also regard progress=1 and missing table in tiflash_indexes as index build is finished, because progress=1 already means TiFlash is aware of this table, and then no entry in tiflash_indexes means exactly this case?

No, we can not tell the difference between these two cases
* A table without any vector index but the tiflash replica is already built
* A table with vector index and the tiflash replica is already built, but tiflash is not aware that the vector index has been added

File: dbms/src/Storages/StorageDeltaMerge.cpp [hunk 2]
```diff
@@ -1394,16 +1398,34 @@ void StorageDeltaMerge::alterSchemaChange(
...
             _store->applySchemaChanges(table_info);
         }
-        else // it seems we will never come into this branch ?
+        else
         {
-            updateTableColumnInfo();
+            if (containsLocalIndexInfos(table_info, log))
+            {
+                // If there exist vector index, then we must init the store to create
+                // at least 1 segment. So that tidb can detect the index is added.
```
[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply dbms/src/Storages/StorageDeltaMerge.cpp:1413]: Actually there is no segment created until there are rows inserted into the DeltaMergeStore instance

**Comment by Lloyd-Pottiger** [comment]
> @Lloyd-Pottiger `DeltaMergeStore::getLocalIndexStats` requires a `DeltaMergeStore` instance to get the stats of local indexes. Without that result, we are hard to tell wihch index_id need to be added to `dt_local_index`
> 
> https://github.com/pingcap/tiflash/blob/1a781c7b0a34e7b87a47770cab5c169d3821ef54/dbms/src/Storages/System/StorageSystemDTLocalIndexes.cpp#L91-L98

The index info can get from table info, and the rows indexed is 0.

**Comment by JaySon-Huang** [comment]
@Lloyd-Pottiger @JinheLin PTAL

**Review by Lloyd-Pottiger — APPROVED** [review]
lgtm

**Comment by Lloyd-Pottiger** [comment]
Need to update description.

**Comment by JaySon-Huang** [comment]
> Need to update description.

Updated

**Pull request merged**